File: Row with max 1s - GFG/row-with-max-1s.cpp
// { Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
//User function template for C++
class Solution{
public:
    int psn(vector<int> &row){
        int l = 0, h = row.size()-1,m;
        while(l<= h){
            m = (l+h)/2;
            
            if(row[m] == 0) l = m+1;
            else h = m-1;
        }
        int x = row.size() - l;
        return x;
    }
	int rowWithMax1s(vector<vector<int>> arr, int n, int m) {
	    // code here
	    int curr = 0, r = -1;
	    for(int i = 0; i<n; i++){
	        int x = psn(arr[i]);
	        if(curr < x){
	            curr = x;
	            r = i;
	        }
	    }
	    return r;
	}

};

// { Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int n, m;
        cin >> n >> m;
        vector< vector<int> > arr(n,vector<int>(m));
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                cin>>arr[i][j];
            }
        }
        Solution ob;
        auto ans = ob.rowWithMax1s(arr, n, m);
        cout << ans << "\n";
    }
    return 0;
}
  // } Driver Code Ends

File: Row with max 1s - GFG/row-with-max-1s.py
#User function Template for python3
class Solution:

	def rowWithMax1s(self,arr, n, m):
		# code here
		def count(nums):
		    l = 0
		    h = m-1
		    
		    while l <= h:
		        mid = (l+h) >> 1
		        if nums[mid] == 0:
		            l = mid + 1
		        else :
		            h = mid - 1
		    return m - l
		maxcount = 0
		rownumber = -1
		for i in range(n):
		    cnt = count(arr[i])
		    if cnt > maxcount:
		        maxcount = cnt
		        rownumber = i
		
		return rownumber

#{ 
#  Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n, m = list(map(int, input().strip().split()))
        
        inputLine = list(map(int, input().strip().split()))
        arr = [[0 for j in range(m)] for i in range(n)]
        
        for i in range(n):
            for j in range(m):
                arr[i][j] = inputLine[i * m + j]
        ob = Solution()
        ans = ob.rowWithMax1s(arr, n, m)
        print(ans)
        tc -= 1

# } Driver Code Ends